### PR TITLE
[testing/integration] - Add test cases covering unprivileged mode for nginx.stubstatus metricset

### PR DESCRIPTION
**Pull request by VihasMakwana**
- Enhancement

## What does this PR do?

- This PR adds test cases covering `unprivileged` mode for nginx metrics integration

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

- Closes https://github.com/elastic/elastic-agent/issues/4818

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: testing/integration/nginx_unprivileged_test.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	compDebugName := ""
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(ctx)
+		if err != nil {
+			runner.T().Logf("agent status returned an error: %v", err)
+			return false
+		}
+
+		for _, comp := range status.Components {
```
Comment (blakerouse) [inline testing/integration/nginx_unprivileged_test.go:149]: Possible to be healthy without any components, should check that components list is not empty and if it is mark it not healthy.

File: testing/integration/nginx_unprivileged_test.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+		healthCheckRefreshTime: time.Second * 5,
+	}
+
+	suite.Run(t, runner)
+}
+
+func (runner *NginxUnprivilegedRunner) SetupSuite() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, "apt-get", "install", "-v", "nginx-full")
```
Comment (blakerouse) [inline testing/integration/nginx_unprivileged_test.go:88]: This is not distro agnostic. You are making this debian specific, so you need to define this in the `define.Require` above.

See above.

File: testing/integration/nginx_unprivileged_test.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	healthCheckTime        time.Duration
+	healthCheckRefreshTime time.Duration
+}
+
+func TestNginxUnprivileged(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: "fleet",
+		Stack: &define.Stack{},
+		OS: []define.OS{
+			{Type: define.Linux},
```
Comment (blakerouse) [inline testing/integration/nginx_unprivileged_test.go:70]:
```suggestion
			{Type: define.Linux, Distro: "ubuntu"},
```

**Review by blakerouse — COMMENTED** [review]
I like these tests, but why are we placing this in the Elastic Agent repository and not in the integration repository?

File: testing/integration/nginx_unprivileged_test.go (hunk 1 above)
Comment (cmacknz) [reply testing/integration/nginx_unprivileged_test.go:149]: It's also possible to be healthy while not yet having read any documents that would require accessing potentially privileged sources.

Just checking health doesn't actually guarantee we tested anything. You need to make sure we ingested data.

Potentially this entire test could be done in https://github.com/elastic/integrations/tree/main/packages/nginx, as long as we have a way to start unprivileged containers. The new Wolfi container would satisfy this by default. 

https://github.com/elastic/integrations/pull/10969 makes the switch to Wolfi. We may get significantly better test coverage by adding this health check to the end of every integration system test, with the advantage that we also won't be exposed to problems in the nginx integration here.

Comment (VihasMakwana) [reply testing/integration/nginx_unprivileged_test.go:149]: > Just checking health doesn't actually guarantee we tested anything. You need to make sure we ingested data.

Sure. Let me add that step.

Comment (blakerouse) [reply testing/integration/nginx_unprivileged_test.go]: Yeah I don't think we should have this test in our repository. I think this is ultimately a test for github.com/elastic/integrations repository.

**Comment by VihasMakwana** [comment]
Closing this. See https://github.com/elastic/elastic-agent/issues/4818#issuecomment-2334564088

**Pull request closed without merging**